File: 2017/test/day_11_test.exs
defmodule AdventOfCode2017.Day11Test do
  use ExUnit.Case

  import AdventOfCode2017.Day11
  import AdventOfCode2017.TestHelper

  test "parse" do
    assert parse(sample_input_1()) == ["ne", "ne", "ne"]
    assert parse(sample_input_2()) == ["ne", "ne", "sw", "sw"]
    assert parse(sample_input_3()) == ["ne", "ne", "s", "s"]
    assert parse(sample_input_4()) == ["se", "sw", "se", "sw", "sw"]
  end

  test "part1" do
    assert part1(sample_input_1()) == 3
    assert part1(sample_input_2()) == 0
    assert part1(sample_input_3()) == 2
    assert part1(sample_input_4()) == 3
    assert part1(input("test/input/day_11.txt")) == 675
  end

  test "part2" do
    assert part2(input("test/input/day_11.txt")) == 1424
  end


  defp sample_input_1, do: "ne,ne,ne"
  defp sample_input_2, do: "ne,ne,sw,sw"
  defp sample_input_3, do: "ne,ne,s,s"
  defp sample_input_4, do: "se,sw,se,sw,sw"
end

File: 2018/test/day_11_test.rb
require "test_helper"
require "day_11"

class Day11Test < Minitest::Test
  def setup
    @day_11 = Day11.new
  end

  def test_part1
    assert_equal "33,45", @day_11.part1(sample_input_1)
    assert_equal "21,61", @day_11.part1(sample_input_2)
    assert_equal "243,49", @day_11.part1(input("day_11.txt"))
    # assert_equal [[33, 45], 29], @day_11.part1(sample_input_1)
    # assert_equal [[21, 61], 30], @day_11.part1(sample_input_2)
    # assert_equal [[243, 49], 31], @day_11.part1(input("day_11.txt"))
  end

  def test_part2
    assert_equal "90,269,16", @day_11.part2(sample_input_1)
    # assert_equal "232,251,12", @day_11.part2(sample_input_2)
    # assert_equal "243,49", @day_11.part2(input("day_11.txt"))
  end

  def test_power_level
    assert_equal 4, @day_11.power_level([3, 5], 8)
    assert_equal -5, @day_11.power_level([122, 79], 57)
    assert_equal 0, @day_11.power_level([217, 196], 39)
    assert_equal 4, @day_11.power_level([101, 153], 71)
  end

  def sample_input_1
    "18"
  end

  def sample_input_2
    "42"
  end
end
